tests/functional/metadata/mongodb/putObject.spec.js

```diff
@@ -0,0 +1,442 @@
+            }),
+            next => {
+                // Disabling versionning and putting new version
+                params.versioning = false;
+                params.versionId = '';
+                return metadata.putObjectMD(bucketName, objName, objVal, params, logger, err => {
+                    if (err) {
+                        return next(err);
+                    }
+                    return next();
+                });
+            },
+            // Master must be updated
+            next => getObject(objName, null, BUCKET_VFORMAT_V0, (err, object) => {
+                assert.deepStrictEqual(err, null);
+                assert.strictEqual(object.key, OBJECT_NAME);
+                assert.notStrictEqual(object.versionId, versionId);
+                return next();
+            }),
+            // The second put shouldn't create a new version
+            next => getObjectCount((err, count) => {
+                assert.deepStrictEqual(err, null);
+                assert.strictEqual(count, 2);
+                return next();
+            }),
+        ], done);
+    });
+
+    it('Should update latest version and repair master', done => {
+        const bucketName = BUCKET_NAME;
+        const objName = OBJECT_NAME;
+        const objVal = {
+            key: OBJECT_NAME,
+            versionId: VERSION_ID,
+            updated: false,
+        };
+        const params = {
+            versioning: true,
+            versionId: VERSION_ID,
+            repairMaster: null,
+        };
+        async.waterfall([
+            // We first create a new version and master
+            next => metadata.putObjectMD(bucketName, objName, objVal, params, logger, err => {
+                if (err) {
+                    return next(err);
+                }
+                return next();
+            }),
+            next => {
+                // Updating the version and repairing master
+                params.repairMaster = true;
+                objVal.updated = true;
+                return metadata.putObjectMD(bucketName, objName, objVal, params, logger, err => {
+                    if (err) {
+                        return next(err);
+                    }
+                    return next();
+                });
+            },
+            // Master must be updated
+            next => getObject(objName, null, BUCKET_VFORMAT_V0, (err, object) => {
+                assert.deepStrictEqual(err, null);
+                assert.strictEqual(object.key, OBJECT_NAME);
+                assert.strictEqual(object.versionId, VERSION_ID);
+                assert.strictEqual(object.updated, true);
+                return next();
+            }),
+            // The second put shouldn't create a new version
+            next => getObjectCount((err, count) => {
+                assert.deepStrictEqual(err, null);
+                assert.strictEqual(count, 2);
+                return next();
+            }),
+        ], done);
+    });
+});
```

Comment:
missing new line at EOF

Reply:
fixed in https://github.com/scality/Arsenal/pull/1729/commits/a6f48b9f342344e6df5ca2e3b7a2d03d8737fbba